keras_nlp/tokenizers/word_piece_tokenizer_trainer.py

```diff
@@ -0,0 +1,147 @@
+
+    Returns:
+        Returns a list of vocabulary terms.
+
+    Examples:
+
+    Basic Usage (from Dataset).
+    >>> inputs = tf.data.Dataset.from_tensor_slices(["bat sat pat mat rat"])
+    >>> vocab = compute_word_piece_vocabulary(inputs, 13)
+    >>> vocab
+    ['[PAD]', '[CLS]', '[SEP]', '[UNK]', '[MASK]', 'a', 'b', 'm', 'p', 'r', 's', 't', '##at']
+    >>> tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(vocabulary=vocab, oov_token="[UNK]")
+    >>> outputs = inputs.map(tokenizer.tokenize)
+    >>> for x in outputs: print(x)
+    tf.Tensor([ 6 12 10 12  8 12  7 12  9 12], shape=(10,), dtype=int32)
+
+    Basic Usage (from filenames).
+    >>> with open("test.txt", "w+") as f: f.write("bat sat pat mat rat\n")
+    >>> inputs = ["test.txt"]
+    >>> vocab = compute_word_piece_vocabulary(inputs, 13)
+
+    Custom Split Usage (from Dataset).
+    >>> def normalize_and_split(x):
+    ...     "Strip punctuation and split on whitespace."
+    ...     x = tf.strings.regex_replace(x, r"\p{P}", "")
+    ...     return tf.strings.split(x)
+    >>> inputs = tf.data.Dataset.from_tensor_slices(["bat sat: pat mat rat.\n"])
+    >>> split_inputs = inputs.map(normalize_and_split)
+    >>> vocab = compute_word_piece_vocabulary(
+    ...     split_inputs, 13, split=False
+    ... )
+    >>> vocab
+    ['[PAD]', '[CLS]', '[SEP]', '[UNK]', '[MASK]', 'a', 'b', 'm', 'p', 'r', 's', 't', '##at']
+    >>> tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(vocabulary=vocab)
+    >>> inputs.map(tokenizer.tokenize)
+
+    Custom Split Usage (from filenames).
+    >>> def normalize_and_split(x):
+    ...     "Strip punctuation and split on whitespace."
+    ...     x = tf.strings.regex_replace(x, r"\p{P}", "")
+    ...     return tf.strings.split(x)
+    >>> with open("test.txt", "w+") as f: f.write("bat sat: pat mat rat.\n")
+    >>> inputs = tf.data.TextLineDataset(["test.txt"])
+    >>> split_inputs = inputs.map(normalize_and_split)
+    >>> vocab = compute_word_piece_vocabulary(split_inputs, 13, split=False)
+    >>> vocab
+    ['[PAD]', '[CLS]', '[SEP]', '[UNK]', '[MASK]', 'a', 'b', 'm', 'p', 'r', 's', 't', '##at']
+    >>> tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(vocabulary=vocab)
+    >>> inputs.map(tokenizer.tokenize)
+    """
+    # Read data files.
+
+    if not isinstance(data, (list, tf.data.Dataset)):
+        raise ValueError(
+            "The `data` argument must be either `tf.data.Dataset` or `list`. "
+            f"Recieved: {type(data)}."
+        )
+    if isinstance(data, list):
+        if not split:
+            raise ValueError(
+                "When learning a vocab from files, `split` must be `True`. "
+                "To compute a vocabulary with custom split rules, load your "
+                "data as a dataset, split it, and pass it to "
+                "`compute_word_piece_vocabulary()` with split=False."
+            )
+        data = tf.data.TextLineDataset(data)
+
+    words_data = data.map(
```

Comment:
Is this only required when `split == True`?

Reply:
It the other way around, tf.data.TextLineDataset(filenames) requires split == True, since we currently do not support reading from filenames and not splitting

Reply:
Yeah me and Jesse talked offline, the supported case for splitting a file with a custom splitter is to load the file yourself and then split yourself before calling this function. This will simply our implementation and recommendations greatly.

Reply:
I expanded the docstring for the helper function just in case, but let me know if you'll like me to remove it